modules/ppcp-blocks/resources/js/card-fields-config.js

```diff
@@ -0,0 +1,44 @@
+const config = wc.wcSettings.getSetting('ppcp-credit-card-gateway_data');
+
+export async function createOrder() {
+    return fetch(config.scriptData.ajax.create_order.endpoint, {
+        method: "POST",
+        headers: {
+            "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+            nonce: config.scriptData.ajax.create_order.nonce,
+            context: config.scriptData.context,
+            payment_method: 'ppcp-credit-card-gateway',
+            save_payment_method: localStorage.getItem('ppcp-save-card-payment') === 'true',
+        }),
+    })
+        .then((response) => response.json())
+        .then((order) => {
+            return order.data.id;
+        })
+        .catch((err) => {
+            console.error(err);
+        });
+}
+
+export function onApprove(data) {
+    fetch(config.scriptData.ajax.approve_order.endpoint, {
```

Comment:
Idea 2 (possibly better):

Maybe just returning a promise here will solve the problem.

```js
export function async onApprove(data) {
    return fetch(...)
}
```

The sample implementation of `onApprove` in the PayPal repository suggests that the `<PayPalCardFieldsProvider>` can accept a Promise as `onApprove` property:
https://github.com/paypal/react-paypal-js/blob/672a48de49ddaea40b5db0f7f0d9c0670d8e7f88/src/stories/utils.ts#L100

Reply:
@stracker-phil just checked and it works like a charm returning a promise, thanks!